### ISSUE TITLE
Implement Eq and Hash traits for Subspace

### ISSUE THREAD

**Issue by CleverAkanoa** [foundationdb-rs/foundationdb-rs#57]
Hello 😀

I'm currently working on a deserializer without parsing, I need to create a Map<Subspace, Element>.

But because Eq + Hash traits aren't bound to Subspace I need to wrap it into a Wrapper(Subspace) structure to satisfy the bound.

Is it possible to implement both of these traits ?

Thanks.

**Comment by PierreZ** [comment]
nice feature :rocket: 

Do you want to implement it?

**Comment by CleverAkanoa** [comment]
I've already implemented  a draft, but it's not tested yet:

```rust
impl PartialEq for Subspace {
    fn eq(&self, other: &Self) -> bool {
        self.bytes() == other.bytes()
    }
}

impl Eq for Wrapper {}

impl Hash for Wrapper {
    fn hash<H: Hasher>(&self, state: &mut H) {
        self.bytes().hash(state);
        state.finish();
    }
}
```

**Comment by PierreZ** [comment]
I think you can directly derive `PartialEq`, `Eq` and `Hash`

**Comment by Speedy37** [comment]
And when implementing Hash, don't call `state.finish()`

**Comment by PierreZ** [comment]
Closed with #62
